sym_api_client_python/datafeed_event_service.py

```diff
@@ -5,358 +5,63 @@
 import logging
 import datetime
 from collections import namedtuple
-import time
 
-from .auth.auth_endpoint_constants import auth_endpoint_constants
-from .exceptions.ServerErrorException import ServerErrorException
+#TODO: These imports are duplicated over the abstract class to avoid errors in the Async version
 from .exceptions.UnauthorizedException import UnauthorizedException
 from .exceptions.APIClientErrorException import APIClientErrorException
 from .exceptions.DatafeedExpiredException import DatafeedExpiredException
 from .exceptions.ServerErrorException import ServerErrorException
 from .exceptions.MaxRetryException import MaxRetryException
 
+from .auth.auth_endpoint_constants import auth_endpoint_constants
+from .services.abstract_datafeed_event_service import AbstractDatafeedEventService
+from .clients.constants.DatafeedVersion import DatafeedVersion
+from .services.datafeed_event_service_v1 import DataFeedEventServiceV1
+from .services.datafeed_event_service_v2 import DataFeedEventServiceV2
+
 log = logging.getLogger(__name__)
 
+
 def make_datetime(unix_timestamp_millis):
     seconds, millis = divmod(unix_timestamp_millis, 1000)
-    return datetime.datetime.utcfromtimestamp(seconds).replace(microsecond=millis*1000)
+    return datetime.datetime.utcfromtimestamp(seconds).replace(microsecond=millis * 1000)
+
 
 EventTrace = namedtuple('EventTrace', 'message_id creation_time bot_received listeners_complete bot_complete')
 
-class DataFeedEventService:
+
+class DataFeedEventService():
```

Comment:
Do you think if it should be inherited from `AbstractDatafeedEventService`?

Reply:
Not really as it's only going to create the appropriate Datafeed Event service class V1 or V2 and those are going to inherit the Abstract Datafeed Event Service class.

Reply:
I will however add methods calling the methods on the DF event service V1 or V2 in the DF event service class.
`def add_room_listener(room_listener):
    self.datafeed_event_service.add_room_listener(room_listener);
    `